### ISSUE TITLE
Display which integration is causing the agent to become unhealthy

### ISSUE THREAD

**Issue by jlind23** [elastic/beats#29442]

**Comment by jlind23** [comment]
ping @ph @cmacknz this is a design tasks to prepare the 8.3 implementation as it's a cross topic.

**Comment by ph** [comment]
I think there are 3 parts:

1. Is to allow the elastic agent to have the state of each inputs and return them back. This is part of the v2 control protocol.
2. Is to modify the input to correctly record their state, this requires changes in the inputs framework, so maybe on data plane side?
3. Is to changes Fleet status to map multiples inputs to a single integrations.

**Comment by ph** [comment]
@blakerouse and I have been discussing 1.

**Comment by cmacknz** [comment]
> 2. Is to modify the input to correctly record their state, this requires changes in the inputs framework, so maybe on data plane side?

Yes, I think this is (or will be) part of the input V2 proposal.
